File: src/R/collatz/col.r
collatz <- function(n)
{
    c <- 0
    while (n != 1)
    {
        if (n %% 2 == 0)
        {
            n = n / 2
        }
        else
        {
            n = n * 3 + 1
        }
        c  = c + 1
    }
    return(c)
}

main <- function()
{
    f <- 0

    for (j in 0:100)
    {
        for (i in 1:100000)
        {
            f = f + collatz(i)
        }
    }

    print(f)
}

main()

File: src/Go/collatz/col.go
package main

import (
    "fmt"
)

func main() {
    f := 0

    for j := 0; j < 100; j++ {
        for i := 1; i <= 100000; i++ {
            f = f + collatz(i)
        }
    }

    fmt.Println(f)
}

func collatz(n int) int {
    c := 0

    for n > 1 {
        if n % 2 == 0 {
            n = n / 2
        } else {
            n = 3 * n + 1
        }
        c = c + 1
    }

    return c
}
